roles/ceph-config/tasks/main.yml

```diff
@@ -9,6 +9,36 @@
       group: "ceph"
       mode: "0755"
 
+  - name: count number of osds for non-lvm scenario
+    set_fact:
+      num_osds: "{{ devices | length }}"
+    when:
+      - devices|length > 0
+      - (osd_scenario == 'collocated' or osd_scenario == 'non-collocated')
+
+  - name: count number of osds for lvm scenario
+    set_fact:
+      num_osds: "{{ lvm_volumes | length }}"
+    when:
+      - lvm_volumes|length > 0
+      - osd_scenario == 'lvm'
+
+  - name: get number of osds for lvm-batch scenario
+    command: "ceph-volume lvm batch --report --format=json --osds-per-device osds_per_device {{ devices| join(' ') }}"
```

Comment:
@andrewschoen is this supposed to be run only in non containerized deployment?

Reply:
I don't know, but I suspect it should work on containerized deployments as well. ``docker_exec_cmd`` is expecting a mon at this point, right?

Reply:
this is exactly what i'm raising this.
It means, we would need to set a new fact like `docker_exec_cmd_osd`

Reply:
@neha-ojha you have to add this task before L26

```
- name: set_fact docker_exec_cmd_osd
  set_fact:
    docker_exec_cmd_osd: "docker exec ceph-osd-{{ hostvars[groups[osd_group_name][0]]['ansible_hostname'] }}"
  when:
    - containerized_deployment
    - groups.get(osd_group_name, []) | length > 0
```

then 

`command: "{{ docker_exec_cmd_osd | default('') }} ceph-volume lvm batch --report --format=json --osds-per-device osds_per_device {{ devices| join(' ') }}"`

Reply:
@guits will this `command:` be added as a part of a new task for `containerized_deployment`or will it replace https://github.com/ceph/ceph-ansible/pull/3113/files#diff-408c9554f34e961d163f68b645ee6da3R27?

Reply:
@guits updated it, let me know if it looks alright. Managed to get https://github.com/ceph/ceph-ansible/pull/3113/commits/5d9f269798c995aeb3baec22744ef0a5da27c7d4#diff-2857b72f37476e1b5d16e7e81783ecbaL241 as a by-product of that change.